components/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/service/EventSimulatorMap.java

```diff
@@ -32,15 +32,17 @@
 
 import java.nio.file.Paths;
 import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 /**
  * EventSimulatorMap holds the simulators available
  */
 public class EventSimulatorMap {
     private static final Logger log = LoggerFactory.getLogger(EventSimulatorMap.class);
     private static final EventSimulatorMap instance = new EventSimulatorMap();
-    private final HashMap<String, ActiveSimulatorData> activeSimulatorMap = new HashMap<>();
-    private final HashMap<String, ResourceDependencyData> inActiveSimulatorMap = new HashMap<>();
+    private final Map<String, ActiveSimulatorData> activeSimulatorMap = new ConcurrentHashMap<>();
```

Comment:
eg: 
  /**
     * Key - name of the model
     * Value - @{@link PerceptronModel}
     */
    private Map<String, PerceptronModel> perceptronModelMap = new HashMap();